garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,164 @@ local meta = FindMetaTable( "Vector" )
 -- Nothing in here, still leaving this file here just in case
 
 --[[---------------------------------------------------------
-	Converts Vector To Color - alpha precision lost, must reset
+Converts Vector to Color - alpha precision lost, must reset
 -----------------------------------------------------------]]
-function meta:ToColor( )
+function meta:ToColor()
+	return Color(self.x * 255, self.y * 255, self.z * 255)
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Table - removes all metrods from the copy
+-----------------------------------------------------------]]
+function meta:ToTable(arr)
+	if(arr) then return {self.x, self.y, self.z} end
+	return {x = self.x, y = self.y, z = self.z}
+end
+
+--[[---------------------------------------------------------
+Converts Vector to Angle - data is copied
+-----------------------------------------------------------]]
+function meta:ToAngle(arr)
+	return Angle(self.x, self.y, self.z)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a rotated vector. Kind of like v:GetNormalized()
+-----------------------------------------------------------]]
+function meta:GetRotated(ang)
+	local v = Vector(self)
+	v:Rotate(ang)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of an added vector
+-----------------------------------------------------------]]
+function meta:GetAdd(vec)
+	local v = Vector(self)
+	v:Add(vec)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a subracted vector
+-----------------------------------------------------------]]
+function meta:GetSub(vec)
+	local v = Vector(self)
+	v:Sub(vec)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a multiplied vector
+-----------------------------------------------------------]]
+function meta:GetMul(num)
+	local v = Vector(self)
+	v:Mul(num)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a divided vector
+-----------------------------------------------------------]]
+function meta:GetDiv(num)
+	local v = Vector(self)
+	v:Div(num)
+	return v
+end
 
-	return Color( self.x * 255, self.y * 255, self.z * 255 )
+--[[---------------------------------------------------------
+Convets the vector to bisector of two vectors
+-----------------------------------------------------------]]
+function meta:Bisect(vec)
+	ns, nv = self:Length(), vec:Length()
+	self:Mul(nv)
+	self:Add(vec:GetMul(ns))
+	return self
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:Bisect(vec)

```

Comment:
Typo

Reply:
Where?

Reply:
Look 10 lines up

Reply:
Ten lines up is
```
--[[---------------------------------------------------------
Convets the vector to bisector of two vectors
-----------------------------------------------------------]]
function meta:Bisect(vec)
```

Reply:
`GetBisect` instead.

Reply:
GetBisect

Reply:
It seems correct o me: https://www.mathsisfun.com/geometry/bisect.html

Reply:
It's still not fixed, dummy